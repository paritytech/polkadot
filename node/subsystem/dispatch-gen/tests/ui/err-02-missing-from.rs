#![allow(dead_code)]

use polkadot_procmacro_subsystem_dispatch_gen::subsystem_dispatch_gen;

/// The event type in question.
#[derive(Clone, Copy, Debug)]
enum Event {
	Smth,
	Else,
}

impl Event {
	fn focus(&self) -> std::result::Result<Intermediate, ()> {
		Ok(Intermediate(self.clone()))
	}
}

#[derive(Debug, Clone)]
struct Intermediate(Event);


/// This should have a `From<Event>` impl but does not.
#[derive(Debug, Clone)]
enum Inner {
	Foo,
	Bar(Intermediate),
}

#[subsystem_dispatch_gen(Event)]
#[derive(Clone)]
enum AllMessages {
	/// Foo
	Vvvvvv(Inner),

    #[skip]
    Uuuuu,
}

fn main() {
    let _x = AllMessages::dispatch_iter(Event::Else);
}
