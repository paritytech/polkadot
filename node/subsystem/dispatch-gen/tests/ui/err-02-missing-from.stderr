error[E0308]: mismatched types
  --> $DIR/err-02-missing-from.rs:29:1
   |
29 | #[subsystem_dispatch_gen(Event)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected enum `Inner`, found struct `Intermediate`
   | help: try using a variant of the expected enum: `Inner::Bar(#[subsystem_dispatch_gen(Event)])`
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
